fs/littlefs/lfs_vfs.c

```diff
@@ -816,16 +834,31 @@ static int littlefs_fchstat(FAR const struct file *filep,
   inode = filep->f_inode;
   fs    = inode->i_private;
 
+  path = lib_get_pathbuffer();
+  if (path == NULL)
+    {
+      return -ENOMEM;
+    }
+
   /* Call LFS to get file size */
 
+  ret = littlefs_convert_result(lfs_file_path(&fs->lfs, &priv->file, path,
```

Comment:
as you know, unfortunately lfs_file_path is another local patch which is not likely accepted by the upstream.
IMO, we should eventually remove the patch as well.
(for other reviewers who don't remember the context; see the discussion in https://github.com/littlefs-project/littlefs/pull/976)

Reply:
before the similar function are added, please don't revert the function, used by other people.

Reply:
IMO, it isn't ok to prevent other people from using other version of littlefs by adding local patches either.
the patch should not have been added in the first place.

Reply:
You can add the combability code/config to math your needed, but simply revert the code used by other community member isn't acceptable.

Reply:
well, i have even suggested an alternative way to implement it months ago. (https://github.com/apache/nuttx/pull/11724#issuecomment-2116644633)
are you suggesting it's my responsibility to actually implement it? why?
it's up to the people who added the code, not me, IMO.

i can understand the ability to use the other version of littlefs is not important for some people.
but for me it's more important than these features.

Reply:
> well, i have even suggested an alternative way to implement it months ago. ([#11724 (comment)](https://github.com/apache/nuttx/pull/11724#issuecomment-2116644633)) are you suggesting it's my responsibility to actually implement it? why? it's up to the people who added the code, not me, IMO.

Yes, since this version is built and tested by ci for a long time ago, and not only used by Xiaomi, but other community member. If you want to use another version of littlefs, it's your responsibility to adapter the mainline code to work with that version.

> 
> i can understand the ability to use the other version of littlefs is not important for some people. but for me it's more important than these features.

Yes, the different team and company has the different interesting. The community version needs more work to satisfy all party.

Reply:
> > well, i have even suggested an alternative way to implement it months ago. ([#11724 (comment)](https://github.com/apache/nuttx/pull/11724#issuecomment-2116644633)) are you suggesting it's my responsibility to actually implement it? why? it's up to the people who added the code, not me, IMO.
> 
> Yes, since this version is built and tested by ci for a long time ago, and not only used by Xiaomi, but other community member. If you want to use another version of littlefs, it's your responsibility to adapter the mainline code to work with that version.

sorry, i don't understand your logic.
i have been using littlefs since before these local patches broke my usage.

> > i can understand the ability to use the other version of littlefs is not important for some people. but for me it's more important than these features.
> 
> Yes, the different team and company has the different interesting. The community version needs more work to satisfy all party.

all parties, but not including me for some reasons?

Reply:
Include, but we need make all interesting party happy, that's why @crafcat7 take time to refine the patch to reduce the change as small as possible.

Reply:
thank you for that.
however, it isn't an appropriate way to fix regressions.
it's more appropriate to do:
1. revert the problematic change
2. take your time to fix it
3. submit reapply with a fix

Reply:
It isn't appreciated also to revert the patch which work well with the littlefs version downloaded by and tested by ci since it break other people's case too.

Reply:
sigh. let me repeat my claim again. these local patches should not have been merged in the first place because they broke "other people's cases".
please don't blame me about reporting regressions and submitted reverts. please blame people who introduced the regressions instead.

anyway, once the next version of littlefs is released, i guess we can use https://github.com/littlefs-project/littlefs/pull/1004 to replace lfs_util.patch.
at that point, i guess we can provide a switch between:
* 2.5.1 with nuttx local patches
* 2.9.4 with no patches

how do you think?

Reply:
> sigh. let me repeat my claim again. these local patches should not have been merged in the first place

https://github.com/apache/nuttx/blob/master/INVIOLABLES.md doesn't prohibit to patch the 3rd project. Sometimes, it's necessary to patch their code.

> because they broke "other people's cases". 

Reverting patch also broke "other people's cases" too. @crafcat7 create https://github.com/apache/nuttx/pull/11729 because @jturnsek ask this feature.

> please don't blame me about reporting regressions and submitted reverts. please blame people who introduced the regressions instead.
> 

Sorry, I don't think that patch the code is a regression.

> anyway, once the next version of littlefs is released, i guess we can use [littlefs-project/littlefs#1004](https://github.com/littlefs-project/littlefs/pull/1004) to replace lfs_util.patch. at that point, i guess we can provide a switch between:
> 
> * 2.5.1 with nuttx local patches
> * 2.9.4 with no patches
> 
> how do you think?

yes, my suggestion is always to keep both parties happy.

Reply:
although it was not a kconfig, it has been overridable for years.
https://github.com/apache/nuttx/blob/ccf0b5af0d6ba555f549d8d00531fa4a3a432a1b/fs/littlefs/Make.defs#L44

Reply:
it's reasonable that the mainline code keeps upgrading to the new release of 3rd party project. If somebody want to stick to the old version, he either create a ci test to ensure that other people don't break the combability, or don't upgrade to the new release.

Reply:
i thought someone (xiaomi?) was relying on the specific version. (2.5.1)
do you mean it should be ok to simply "upgrade" the default to 2.9.x?

Reply:
> i thought someone (xiaomi?) was relying on the specific version. (2.5.1) 

Yes, we depend on a specific version of littlefs. But it isn't a reason to block the community upgrade to a new version.

> do you mean it should be ok to simply "upgrade" the default to 2.9.x?

Yes, the mainline should track the last version of 3rd project. @crafcat7 please provide a patch to switch LITTLEFS_VERSION  to the latest one.

Reply:
> anyway, once the next version of littlefs is released, i guess we can use [littlefs-project/littlefs#1004](https://github.com/littlefs-project/littlefs/pull/1004) to replace lfs_util.patch. at that point, i guess we can provide a switch between:
> 
>     * 2.5.1 with nuttx local patches
> 
>     * 2.9.4 with no patches

see https://github.com/apache/nuttx/pull/14956

Reply:
so, I think that this patch doesn't need anymore, let's close it now.